etna/models/linear.py

```diff
@@ -1,11 +1,51 @@
+import numpy as np
+import pandas as pd
 from sklearn.linear_model import ElasticNet
 from sklearn.linear_model import LinearRegression
 
-from etna.models.sklearn import SklearnMultiSegmentModel
-from etna.models.sklearn import SklearnPerSegmentModel
+from etna.models.base import NonPredictionIntervalContextIgnorantAbstractModel
+from etna.models.mixins import MultiSegmentModelMixin
+from etna.models.mixins import NonPredictionIntervalContextIgnorantModelMixin
+from etna.models.mixins import PerSegmentModelMixin
+from etna.models.sklearn import _SklearnAdapter
 
 
-class LinearPerSegmentModel(SklearnPerSegmentModel):
+class _LinearAdapter(_SklearnAdapter):
+    def predict_components(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Estimate prediction components.
+
+        Parameters
+        ----------
+        df:
+            features dataframe
+
+        Returns
+        -------
+        :
+            dataframe with prediction components
+        """
+        if self.regressor_columns is None:
+            raise ValueError("Model is not fitted! Fit the model before estimating forecast components!")
+
+        component_names = [f"target_component_{component_name}" for component_name in self.regressor_columns]
+        target_components = df[self.regressor_columns]
+        components_coefs = self.model.coef_
```

Comment:
Are we really sure that ordering of `self.model.coef_` is aligned with ordering of `self.regressor_columns`?

Reply:
May be we can extract the exact names of columns and use them for alignment.